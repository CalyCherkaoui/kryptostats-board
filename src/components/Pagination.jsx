import React from 'react';
import ReactPaginate from 'react-paginate';
import PropTypes from 'prop-types';

const Pagination = ({ count, fetchData }) => (
  <div className="pagination_container">
    <ReactPaginate
      pageCount={Math.ceil(count / 6)}
      pageRangeDisplayed={1}
      marginPagesDisplayed={2}
      onPageChange={fetchData()}
    />
  </div>
);

Pagination.propTypes = ({
  count: PropTypes.number,
  fetchData: PropTypes.func,
}).isRequired;

export default Pagination;
