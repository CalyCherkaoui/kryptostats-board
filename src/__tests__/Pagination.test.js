// import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Pagination from '../components/Pagination';

describe(
  'Renders correctly The navigation bar component',
  () => {
    const count = 1;
    const fetchData = () => true;

    it(
      'Matches Pagination snapshot',
      () => {
        const rendered = renderer.create(
          <Provider store={store}>
            <Router>
              <Pagination count={count} fetchData={fetchData} />
            </Router>
          </Provider>,
        );

        expect(rendered).toMatchSnapshot();
      },
    );
  },
);
