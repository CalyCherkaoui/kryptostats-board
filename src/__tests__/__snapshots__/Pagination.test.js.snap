// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly The navigation bar component Matches Pagination snapshot 1`] = `
<div
  className="pagination_container"
>
  <ul>
    <li
      className="previous disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Previous page"
        onClick={[Function]}
        onKeyPress={[Function]}
        rel="prev"
        role="button"
        tabIndex="0"
      >
        Previous
      </a>
    </li>
    <li
      className="selected"
    >
      <a
        aria-current="page"
        aria-label="Page 1 is your current page"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex="0"
      >
        1
      </a>
    </li>
    <li
      className="next disabled"
    >
      <a
        aria-disabled="true"
        aria-label="Next page"
        onClick={[Function]}
        onKeyPress={[Function]}
        rel="next"
        role="button"
        tabIndex="0"
      >
        Next
      </a>
    </li>
  </ul>
</div>
`;
